### PR TITLE
Merge master jdk-21.0.4+1 into openj9-staging

### PR DESCRIPTION
**Pull request by JasonFengJ9**
Merge master jdk-21.0.4+1 into openj9-staging

Conflicts:
```
	src/java.base/share/classes/java/lang/VirtualThread.java
	test/hotspot/jtreg/serviceability/jvmti/vthread/GetThreadState/GetThreadStateTest.java
	test/jdk/java/lang/Thread/virtual/ThreadAPI.java
	test/jdk/java/lang/Thread/virtual/stress/ParkALot.java
	test/jdk/java/lang/Thread/virtual/stress/Skynet.java
```

Requires
* https://github.com/eclipse-openj9/openj9/pull/19434

Signed-off-by: Jason Feng <fengj@ca.ibm.com>

### CODE REVIEW

File: test/hotspot/jtreg/serviceability/jvmti/vthread/GetThreadState/GetThreadStateTest.java [hunk 1]
```diff
@@ -25,17 +25,18 @@
...
  * @test id=no-vmcontinuations
  * @requires vm.continuations
+ * @library /test/lib
  * @run junit/othervm/native -XX:+UnlockExperimentalVMOptions -XX:-VMContinuations GetThreadStateTest
  */
 
 import java.util.StringJoiner;
-import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.locks.LockSupport;
```
Comment (keithc-ca) [inline test/hotspot/jtreg/serviceability/jvmti/vthread/GetThreadState/GetThreadStateTest.java:42]: This is missing from upstream:
```java
import jdk.test.lib.thread.VThreadPinner;
```

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: test/hotspot/jtreg/serviceability/jvmti/vthread/GetThreadState/GetThreadStateTest.java (hunk 1 above)
Comment (JasonFengJ9) [reply test/hotspot/jtreg/serviceability/jvmti/vthread/GetThreadState/GetThreadStateTest.java:43]: Restored this `import`.

Comment (keithc-ca) [reply test/hotspot/jtreg/serviceability/jvmti/vthread/GetThreadState/GetThreadStateTest.java:43]: Please put it in the same place as upstream; this file should be identical to the content of 176b7aa48c6a9ba32b450840f69c776f621646f3.

Comment (JasonFengJ9) [reply test/hotspot/jtreg/serviceability/jvmti/vthread/GetThreadState/GetThreadStateTest.java:42]: Matched the upstream.

**Review by keithc-ca — APPROVED** [review]

**Pull request merged**